### PR TITLE
test(css): Lint for presence and use of css modules

### PR DESCRIPTION
**Pull request by kosanna**
## Purpose

Prevent unused and undefined css classes from creeping into the code base by validating the presence and use of css modules.

## Approach

Use `eslint-plugin-css-modules` to lint for the usage and definition of css modules. This ensures nothing gets left behind when refactoring.

Lots of the things are ignored and need to be fixed in the future, at least we will have the tool

Also fixes duplicated content on tablet view on JobCard

### CODE REVIEW

File: react/Alert/Alert.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint css-modules/no-unused-class: [2, { markAsUsed: ['positive', 'info', 'critical', 'help'] }] */
```
Comment (ashin) [inline react/Alert/Alert.js]: This is why this linting rule can be a massive pain.

File: react/Checkbox/Checkbox.demo.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ class CheckboxContainer extends Component {
     const { component: DemoComponent, componentProps } = this.props;
     const { checked } = this.state;
     const classNames = classnames({
-      [demoStyles.root]: true,
+      [demoStyles.root]: true, // eslint-disable-line
```
Comment (ashin) [inline react/Checkbox/Checkbox.demo.js]: Why does this cause an error? Also we should only disable the single rule that is breaking?

File: standalone/header-footer/src/StandaloneProvider/StandaloneProvider.js [hunk 3]
```diff
@@ -31,6 +31,7 @@ export default class StandaloneProvider extends Component {
 
     return (
       <div className={cssSelectorPrefix.replace('.', '')}>
+        {/* eslint-disable-next-line */}
```
Comment (ashin) [inline standalone/header-footer/src/StandaloneProvider/StandaloneProvider.js]: Same as here.

**Comment by ashin** [comment]
The amount of exceptions needed for this rule, really makes me second guess the use of this rule.

What were we trying to prevent? What was the issue that caused this?

File: react/Alert/Alert.js (hunk 1 above)
Comment (kosanna) [reply react/Alert/Alert.js]: This saves us more pain than it adds. All the pain is already in this pr

File: react/Checkbox/Checkbox.demo.js (hunk 2 above)
Comment (kosanna) [reply react/Checkbox/Checkbox.demo.js]: no idea why it can't see the class there, will change to single

**Comment by kosanna** [comment]
Yesterday we had a prod bug when a thing was separated into a subcomponent folder and styles have been forgotten to carry over next to it. This rule would have highlighted it as unused classes in this case in the main folder, which would have made it super obvious that it's not been done properly

File: docs/src/components/Home/Hero/ButtonsPreview/ButtonsPreview.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
+/* eslint css-modules/no-unused-class: [2, { markAsUsed: ['selected', 'rootFocused', 'root', 'compact', 'disabled', 'root_callToAction', 'jobsDB', 'jobStreet', 'root_primary', 'focused', 'root_secondary', 'root_hyperlink', 'root_ghost_white', 'root_tertiary', 'group'] }] */
```
Comment (hscgavin) [inline docs/src/components/Home/Hero/ButtonsPreview/ButtonsPreview.js]: 😮

File: docs/src/components/Home/Hero/ButtonsPreview/ButtonsPreview.js [hunk 5]
```diff
@@ -1,23 +1,13 @@
...
-        <Button color="secondary" className={classnames(styles.button, buttonStyles.rootActive)}>Pressed</Button>
-      </div>
-      <div className={styles.group}>
-        <Button color="primary" className={styles.button}>Normal</Button>
-        <Button color="primary" className={classnames(styles.button, buttonStyles.rootHover)}>Hover</Button>
-        <Button color="primary" className={classnames(styles.button, buttonStyles.rootActive)}>Pressed</Button>
-      </div>
+    <div>
+      <Button color="primary" className={styles.button}>Primary</Button>
+      <Button color="secondary" className={styles.button}>Secondary</Button>
```
Comment (patrickliu-sa) [inline docs/src/components/Home/Hero/ButtonsPreview/ButtonsPreview.js]: Is it possible to map and showcase all available CTA button styles from different tenants together (excluding `seekasia`)?

Comment (kosanna) [reply docs/src/components/Home/Hero/ButtonsPreview/ButtonsPreview.js]: Not the purpose of this PR, but I could take a look at it

Comment (kosanna) [reply docs/src/components/Home/Hero/ButtonsPreview/ButtonsPreview.js]: I can't really change the main provider locally, so let's just leave it as-is here

**Comment by kosanna** [comment]
Let's give it a go and hope it does not break anything in prod. Shouldn't 🤞We need to make sure it's visually reviewed very thoroughly

**Pull request merged**